### PR TITLE
Replace unnecessary dependency in webapi samples

### PR DESCRIPTION
**Pull request by wujack778**
Replace unnecessary dependency in `webapi ` samples
[task link](https://github.com/Azure/azure-sdk-for-java/pull/18412#discussion_r551095639)

### CODE REVIEW

File: sdk/spring/azure-spring-boot-samples/azure-spring-boot-sample-active-directory-resource-server-obo/README.md [hunk 1]
```diff
@@ -41,17 +37,13 @@ We need to ensure that this [environment checklist][ready-to-run-checklist] is c
       <artifactId>spring-security-oauth2-resource-server</artifactId>
     </dependency>
     <dependency>
-      <groupId>org.springframework.security</groupId>
-      <artifactId>spring-security-oauth2-client</artifactId>
+      <groupId>org.springframework.boot</groupId>
+      <artifactId>spring-boot-starter-oauth2-client</artifactId>
```
Comment (chenrujun) [inline sdk/spring/azure-spring-boot-samples/azure-spring-boot-sample-active-directory-resource-server-obo/README.md:41]: 1. Can we delete `spring-boot-starter-security`?
2. Can we delete `msal4j`?
3. Please sort the dependencies, and add comments in pom file to make it easier to read. Like this: https://github.com/chenrujun/azure-sdk-for-java/blob/3e66ff7ec96e8d695b212a1f992c322cfe01c612/sdk/spring/azure-spring-boot-test-aad-obo/pom.xml

Comment (wujack778) [reply sdk/spring/azure-spring-boot-samples/azure-spring-boot-sample-active-directory-resource-server-obo/README.md:41]: get it

**Review by chenrujun — APPROVED** [review]
Hi, @wujack778 , did you test it after you update the pom?

**Comment by wujack778** [comment]
> after

I passed the sample test

**Comment by wujack778** [comment]
I passed the sample test @chenrujun

File: sdk/spring/azure-spring-boot-samples/azure-spring-boot-sample-active-directory-resource-server/README.md [hunk 2]
```diff
@@ -24,12 +24,12 @@ We need to ensure that this [environment checklist][ready-to-run-checklist] is c
       <artifactId>azure-spring-boot-starter-active-directory</artifactId>
     </dependency>
     <dependency>
-      <groupId>org.springframework.security</groupId>
-      <artifactId>spring-security-oauth2-resource-server</artifactId>
+      <groupId>org.springframework.boot</groupId>
+      <artifactId>spring-boot-starter-oauth2-client</artifactId>
```
Comment (saragluna) [inline sdk/spring/azure-spring-boot-samples/azure-spring-boot-sample-active-directory-resource-server/README.md:28]: Why do we need oauth2-client here?

File: sdk/spring/azure-spring-boot-samples/azure-spring-boot-sample-active-directory-resource-server/pom.xml [hunk 3]
```diff
@@ -20,25 +20,19 @@
...
-    <dependency>
-      <groupId>org.springframework.security</groupId>
-      <artifactId>spring-security-oauth2-jose</artifactId>
-    </dependency>
+
+    <!-- spring boot starter dependencies. -->
     <dependency>
       <groupId>org.springframework.boot</groupId>
-      <artifactId>spring-boot-starter-web</artifactId>
+      <artifactId>spring-boot-starter-oauth2-client</artifactId>
```
Comment (saragluna) [inline sdk/spring/azure-spring-boot-samples/azure-spring-boot-sample-active-directory-resource-server/pom.xml:27]: Same here.

**Review by saragluna — CHANGES_REQUESTED** [review]

**Review by chenrujun — APPROVED** [review]

**Review by saragluna — APPROVED** [review]
LGTM

**Comment by saragluna** [comment]
@wujack778 I will merge this PR for now. @moarychan will help create another PR, to: 
- Update the aad starter README, https://github.com/Azure/azure-sdk-for-java/tree/master/sdk/spring/azure-spring-boot-starter-active-directory#include-the-package-1, to remove the include package part for two Web APIs cases.

image
image


- Remove the include packages part for two Web API samples, to make them consistent with our other samples.

**Pull request merged**